src/py/mat3ra/made/tools/utils/functions.py

```diff
@@ -0,0 +1,105 @@
+from typing import List, Literal
+
+import numpy as np
+from pydantic import BaseModel
+from scipy.integrate import quad
+from scipy.optimize import root_scalar
+
+AXIS_TO_INDEX_MAP = {"x": 0, "y": 1, "z": 2}
+EQUATION_RANGE_COEFFICIENT = 5
+
+
+class FunctionHolder(BaseModel):
+    def apply_function(self, coordinate: List[float]) -> float:
+        """
+        Get the value of the function at the given coordinate.
+        """
+        raise NotImplementedError
+
+    def apply_derivative(self, coordinate: List[float]) -> float:
+        """
+        Get the derivative of the function at the given coordinate
+        """
+        raise NotImplementedError
+
+    def get_arc_length(self, a: float, b: float) -> float:
+        """
+        Get the arc length of the function between a and b.
+        """
+        raise NotImplementedError
+
+    def get_json(self) -> dict:
```

Comment:
Let's check for this to be used instead of the tuple logic in all the previously written code also